File: 0965.单值二叉树/0965-单值二叉树.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    bool isUnivalTree(TreeNode* root) {
        if( !root ) {
            return true;
        }
        return dfs(root, root->val);
    }

    bool dfs(TreeNode* root, int value) {
        if( !root ) {
            return true;
        }

        if( root->val != value ) {
            return false;
        }

        return dfs(root->left, value) && dfs(root->right, value);
    }

};

File: 0965.单值二叉树/0965-单值二叉树.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isUnivalTree(self, root: TreeNode) -> bool:
        if None == root:
            return True
        return self.dfs(root, root.val)
    
    def dfs(self, root: TreeNode, val: int) -> bool:
        if None == root:
            return True
        
        if root.val != val:
            return False
        
        return self.dfs(root.left, val) and self.dfs(root.right, val)